src/auto/useNavigationAuto.ts

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { NativeModules } from 'react-native';
+import type { MapViewAutoController, NavigationAutoCallbacks } from './types';
+import { useModuleListeners, type Location } from '../shared';
+import type {
+  MapType,
+  CircleOptions,
+  Circle,
+  MarkerOptions,
+  Marker,
+  PolylineOptions,
+  Polyline,
+  PolygonOptions,
+  Polygon,
+  CameraPosition,
+  UISettings,
+} from '../maps';
+import { useMemo } from 'react';
+
+const { NavAutoEventDispatcher, NavAutoModule } = NativeModules;
+
+const androidBridge: string = 'NavAutoJavascriptBridge';
+
+export const useNavigationAuto = (): {
+  mapViewAutoController: MapViewAutoController;
+  addListeners: (listeners: Partial<NavigationAutoCallbacks>) => void;
+  removeListeners: (listeners: Partial<NavigationAutoCallbacks>) => void;
+  removeAllListeners: () => void;
+} => {
+  const moduleListenersHandler = useModuleListeners<NavigationAutoCallbacks>(
+    NavAutoEventDispatcher,
+    androidBridge,
+    ['onAutoScreenAvailabilityChanged', 'onCustomNavigationAutoEvent']
+  );
+
+  const mapViewAutoController = useMemo(
+    () => ({
+      cleanup: async () => {
+        moduleListenersHandler.removeAllListeners();
+      },
+
+      isAutoScreenAvailable: async () => {
+        return await NavAutoModule.isAutoScreenAvailable();
+      },
+
+      setMapType: (mapType: MapType) => {
+        NavAutoModule.setMapType(mapType);
+      },
+
+      setMapStyle: (mapStyle: string) => {
+        NavAutoModule.setMapStyle(mapStyle);
+      },
+
+      // Android only.
+      setMapToolbarEnabled: (enabled: boolean) => {
```

Comment:
Nit: Is this "Android only" surfacing to consumers?

Reply:
The documentation for appfacing setMapToolbarEnabled method had misleading documentation. Even when it is not part of thie PR, i updated the documentation as well, including the information that this is Android only feature